libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -225,9 +226,17 @@ export class Simulator {
   // keeps the running state of all resources.
   private runningState: Record<string, ResourceRunningState> = {};
 
+  private lockfile: Lockfile;
+
   constructor(props: SimulatorProps) {
     const simdir = resolve(props.simfile);
     this.statedir = props.stateDir ?? join(simdir, ".state");
+    this.lockfile = new Lockfile({
+      path: join(this.statedir, ".lock"),
+      onCompromised: () => {
+        this.stop().catch(() => {});
```

Comment:
This would cause any exceptions raised by stop() to be caught / hidden from the user, I wonder if we should avoid that? Maybe
```suggestion
        void this.stop();
```

Reply:
My intention was to hide problems like "Can't stop simulator if it's not running" but that should never happen in the first place... So good call, will remove that.

Reply:
The simulator now logs these errors too 👍🏻